packages/server/src/fhir/repo.ts

```diff
@@ -523,11 +526,16 @@ export class Repository extends BaseRepository implements FhirRepository {
     return result;
   }
 
-  private validate(resource: Resource): void {
+  private async validate(resource: Resource): Promise<void> {
     if (this.context.strictMode) {
       const start = process.hrtime.bigint();
+      const profileURLs = resource.meta?.profile;
       try {
-        validate(resource);
+        if (profileURLs) {
+          await this.validateProfiles(resource, profileURLs);
+        } else {
```

Comment:
Hm, interesting that this is `else` and not both.

I suppose the operating principle here is that the FHIR Profile `StructureDefinition` is supposed to include everything in the base definition, and therefore validating against the base definition would be redundant?

For FHIR Profiles published by HL7/FHIR, that seems reasonable.  For user submitted profiles, that means we will need a mechanism of validating that the submitted profile is compatible with the base `StructureDefinition`, otherwise this could be a backdoor to all kinds of shenanigans.

Reply:
Yeah, in the expected case the `snapshot` of the profile should contain all base element definitions included as well.  I agree though, it's likely better to be defensive and pay the cost of double validating here rather than trying to bulletproof the process of adding new `StructureDefinition` for use in validation